File: app/src/types/market.ts
export type MarketStatus = 'active' | 'resolved' | 'expired' | 'disputed';
export type MarketOutcome = 'yes' | 'no' | null;

export interface Market {
  id: string;
  pubkey: string;
  creator: string;
  question: string;
  voiceUri?: string;
  category: string;
  region: string;
  isPrivate: boolean;
  isLight?: boolean;
  status: MarketStatus;
  outcome: MarketOutcome;
  yesPool: number;
  noPool: number;
  totalBets: number;
  shieldedBetCount?: number;
  createdAt: number;
  expiresAt: number;
  resolvedAt?: number;
  // oracle-resolved markets (Pyth/Switchboard relay)
  oracleAuthority?: string;
  oracleThreshold?: number;
  oracleAbove?: boolean;
}

export interface Bet {
  id: string;
  marketId: string;
  user: string;
  outcome: 'yes' | 'no';
  amount: number;
  isShielded: boolean;
  timestamp: number;
  claimed: boolean;
}

export interface UserBetPosition {
  bet: Bet | null;
  potentialWinnings: number;
  isWinner: boolean;
}

export interface MarketCategory {
  id: string;
  name: string;
  icon: string;
  color: string;
}

File: programs/ilowa/src/state/market.rs
use anchor_lang::prelude::*;

#[account]
#[derive(InitSpace)]
pub struct Market {
    pub creator: Pubkey,
    #[max_len(280)]
    pub question: String,
    #[max_len(32)]
    pub category: String,
    #[max_len(32)]
    pub region: String,
    pub is_private: bool,
    pub status: MarketStatus,
    pub outcome: Option<bool>,
    pub yes_pool: u64,
    pub no_pool: u64,
    pub total_bets: u32,
    pub created_at: i64,
    pub expires_at: i64,
    pub resolved_at: Option<i64>,
    pub bump: u8,
}

#[derive(AnchorSerialize, AnchorDeserialize, Clone, Copy, PartialEq, Eq, InitSpace)]
pub enum MarketStatus {
    Active,
    Resolved,
    Expired,
    Disputed,
}

#[account]
#[derive(InitSpace)]
pub struct Bet {
    pub market: Pubkey,
    pub user: Pubkey,
    pub outcome: bool,
    pub amount: u64,
    pub is_shielded: bool,
    pub timestamp: i64,
    pub claimed: bool,
    pub bump: u8,
}
